subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/dependencies/DefaultDependencyHandler.java

```diff
@@ -354,8 +357,11 @@ public Dependency platform(Object notation, Action<? super Dependency> configure
     public Dependency enforcedPlatform(Object notation) {
         Dependency platformDependency = create(notation);
         if (platformDependency instanceof ExternalModuleDependency) {
-            ExternalModuleDependency externalModuleDependency = (ExternalModuleDependency) platformDependency;
-            DeprecationLogger.whileDisabled(() -> externalModuleDependency.setForce(true));
+            AbstractExternalModuleDependency externalModuleDependency = (AbstractExternalModuleDependency) platformDependency;
+            externalModuleDependency.setForceForEnforcedPlatform(true);
+            ModuleVersionIdentifier version = DefaultModuleVersionIdentifier.newId(externalModuleDependency.getModule(), externalModuleDependency.getVersion());
+            MutableVersionConstraint constraint = (MutableVersionConstraint) externalModuleDependency.getVersionConstraint();
+            constraint.strictly(version.getVersion());
```

Comment:
Is the force flag still needed if we're setting a "strictly" version?

Reply:
Probably not.  Removed.